processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/TaskRequest.java

```diff
@@ -16,50 +16,37 @@
 
 package com.linecorp.decaton.processor.runtime.internal;
 
-import org.apache.kafka.common.TopicPartition;
-import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
 
 import com.linecorp.decaton.processor.runtime.internal.PerKeyQuotaManager.QuotaUsage;
 import com.linecorp.decaton.processor.tracing.TracingProvider.RecordTraceHandle;
 
 import lombok.AllArgsConstructor;
 import lombok.Getter;
-import lombok.ToString;
 import lombok.experimental.Accessors;
 
-@ToString
 @Getter
 @Accessors(fluent = true)
 @AllArgsConstructor
 public class TaskRequest {
-    private final TopicPartition topicPartition;
-    private final long recordOffset;
     private final OffsetState offsetState;
-    @ToString.Exclude
-    private final byte[] key;
-    @ToString.Exclude
-    private final Headers headers;
-    @ToString.Exclude
+    private final ConsumerRecord<byte[], byte[]> record;
     private final RecordTraceHandle trace;
-    @ToString.Exclude
-    private byte[] rawRequestBytes;
-    @ToString.Exclude
     private final QuotaUsage quotaUsage;
 
     public String id() {
         // TaskRequest object is held alive through associated ProcessingContext's lifetime, hence holding
         // any value as its field makes memory occupation worse. Since this ID field is rarely used (typically
         // when trace level logging is enabled), it is better to take short lived object allocation and cpu cost
         // rather than building it once and cache as an object field.
-        return "topic=" + topicPartition.topic() + " partition=" + topicPartition.partition() +
-               " offset=" + recordOffset;
+        return "topic=" + record.topic() + " partition=" + record.partition() +
+               " offset=" + record.offset();
     }
 
-    /**
-     * This class will live until the task process has been completed.
-     * To lessen heap pressure, rawRequestBytes should be purged by calling this once the task is extracted.
-     */
-    public void purgeRawRequestBytes() {
```

Comment:
I found that purging rawRequestBytes doesn't make much sense because `ProcessingContextImpl.task.taskDataBytes` anyways hold bytes before extraction, and ProcessingContextImpl lives until the task is completed.

The exception is when task is serialized as `DecatonTaskRequest`.
In this case,  `task.taskDataBytes` and `ConsumerRecord#value` differs so purging latter might be effective though, since `DecatonTaskRequest` is already deprecated, we don't need to take care of this case I think.

Reply:
H-m that sounds right.

Let me consider about this again then

Reply:
I reverted the code to still do purging after extraction https://github.com/line/decaton/pull/238/commits/644474b84afa63bcd59b8f677c32ec02a4f28d55